cdap-watchdog/src/main/java/co/cask/cdap/logging/read/AvroFileReader.java

```diff
@@ -181,9 +174,34 @@ public void readLog(Location file, Filter logFilter, long fromTimeMs, long toTim
     }
   }
 
+  private List<LogEvent> readToNextSync(Filter logFilter, long fromTimeMs, DataFileReader<GenericRecord> dataFileReader,
+                             long startPosition, long endPosition) throws IOException {
```

Comment:
I don't think there is a need to pass `startPosition` to this method, since it is just `dataFileReader.previousSync()`, which can be figured out in the method itself.